addons/account/models/account_tax.py

```diff
@@ -1598,6 +1598,85 @@ def _fix_tax_included_price_company(self, price, prod_taxes, line_taxes, company
             line_taxes = line_taxes.filtered(lambda tax: tax.company_id == company_id)
         return self._fix_tax_included_price(price, prod_taxes, line_taxes)
 
+    @api.model
+    def _dispatch_negative_lines_for_global_discount(self, base_lines):
```

Comment:
Plz rename it to `_dispatch_negative_lines`... or be more specific like `_dispatch_negative_lines_for_global_discount_for_reward_line_for_gift_card_for_e_wallet_for_loyalty_for_anything_implying_negative_lines_being_not_a_global_discount`